src/CronExpressionParser.ts

```diff
@@ -0,0 +1,273 @@
+
+    // Replace '*' and '?'
+    value = value.replace(/[*?]/g, constraints.min + '-' + constraints.max);
+    return CronExpressionParser.#parseSequence(value, constraints, field);
+  }
+
+  /**
+   * Parse a sequence from a cron expression.
+   * @param {string} val - The sequence to parse.
+   * @param {CronConstraints} constraints - The constraints for the field.
+   * @param {keyof typeof CronUnit} field - The field to parse.
+   * @private
+   */
+  static #parseSequence(val: string, constraints: CronConstraints, field: keyof typeof CronUnit): (number | string)[] {
+    const stack: (number | string)[] = [];
+
+    function handleResult(result: number | string | (number | string)[], constraints: CronConstraints) {
+      if (Array.isArray(result)) {
+        result.forEach((value) => {
+          /* istanbul ignore else - FIXME no idea how this is triggered or what it's purpose is */
+          if (!CronExpressionParser.#isValidConstraintChar(constraints, value)) {
+            const v = parseInt(value.toString(), 10);
+            const isValid = v >= constraints.min && v <= constraints.max;
+            assert(isValid, `Constraint error, got value ${value} expected range ${constraints.min}-${constraints.max}`);
+            stack.push(value);
+          } else {
+            /* istanbul ignore next - FIXME no idea how this is triggered or what it's purpose is */
+            stack.push(value);
+          }
+        });
+      } else {
+        if (CronExpressionParser.#isValidConstraintChar(constraints, result)) {
+          stack.push(result);
+        } else {
+          const v = parseInt(result.toString(), 10);
+          const isValid = v >= constraints.min && v <= constraints.max;
+          assert(isValid, `Constraint error, got value ${result} expected range ${constraints.min}-${constraints.max}`);
+          stack.push(field === 'DayOfWeek' ? v % 7 : result);
+        }
+      }
+    }
+
+    const atoms = val.split(',');
+    assert(atoms.every((atom) => atom.length > 0), 'Invalid list value format');
+    atoms.forEach((atom) => handleResult(CronExpressionParser.#parseRepeat(atom, constraints, field), constraints));
+    return stack;
+  }
+
+  /**
+   * Parse repeat from a cron expression.
+   * @param {string} val - The repeat to parse.
+   * @param {CronConstraints} constraints - The constraints for the field.
+   * @param {keyof typeof CronUnit} field - The field to parse.
+   * @private
+   * @returns {(number | string)[]} The parsed repeat.
+   */
+  static #parseRepeat(val: string, constraints: CronConstraints, field: keyof typeof CronUnit): ParseRageResponse {
+    const atoms = val.split('/');
+    assert(atoms.length <= 2, `Invalid repeat: ${val}`);
+    if (atoms.length === 2) {
+      if (!isNaN(parseInt(atoms[ 0 ]))) {
+        atoms[ 0 ] = `${atoms[ 0 ]}-${constraints.max}`;
+      }
+      return CronExpressionParser.#parseRange(atoms[ 0 ], parseInt(atoms[ 1 ]), constraints, field);
```

Comment:
```suggestion
      return CronExpressionParser.#parseRange(atoms[ 0 ], parseInt(atoms[ 1 ], 10), constraints, field);
```

Let's set radix when calling `parseInt`. There are some other similar cases when radix is not set.